tests/test_captions.py

```diff
@@ -14,27 +15,59 @@
 
 @pytest.mark.usefixtures('mock_client', 'make_mock_response')
 class TestCaptionEndpoint():
-    def test_get_captions(self, mock_client, make_mock_response):
+    def test_get_captions_default_content_type(self, mock_client, make_mock_response):
         data = 'Test'
+        expected_content_type = CaptionType.SRT.value
         response = make_mock_response(url=URL, text=data)
         mock_client.session.request.return_value = response
 
         res = mock_client.get_captions(JOB_ID)
+        assert res == data
+        mock_client.session.request.assert_called_once_with(
+            "GET",
+            URL,
+            headers={'Accept': expected_content_type}
```

Comment:
Do you remember how we refactored the node sdk to put the HTTP call in another layer so that we could make easier assertions? Should we do something similar here?

It appears that every small change to one of our method that touches an endpoint forces us to make a large number of line edits in the test file.

Reply:
We already pretty much do this in self._make_http_request. It's essentually the same as we have in the python SDK. I don't think it would cut down on tests realistically

Reply:
I think the real cause is that we have `_as_stream` and normal versions of each output method. Essentially doubling the number of tests